pkg/algorithm/cstorpoolselect/v1alpha1/select.go

```diff
@@ -225,53 +459,68 @@ func (s *selection) validate() error {
 // filter returns the final list of pools that
 // gets selected, after passing the original list
 // of pools through the registered selection policies
-func (s *selection) filter() ([]string, error) {
+func (s *selection) filter() (*csp.CSPList, error) {
 	var (
-		filtered []string
-		err      error
+		err error
 	)
-	if len(s.policies) == 0 {
-		return s.poolUIDs, nil
+	if len(s.policies.items) == 0 {
+		return s.pools, nil
 	}
 	// make a copy of original pool UIDs
-	filtered = append(filtered, s.poolUIDs...)
-	for _, policy := range s.policies {
+	filtered := &csp.CSPList{}
```

Comment:
we need to make use of either of below
i.e.
```go
filtered := csp.ListBuilder().WithList(s.pools).List()
// or 
filtered := csp.ListCopy(s.pools)
```

Reply:
done